Detailed the description to enable easy setup

helios1101 opened the pull request:
<!-- Thank you for contributing a pull request! Here are a few tips to help you:

1. If this is your first contribution, make sure you've read the Contribution Checklist https://developers.mattermost.com/contribute/getting-started/contribution-checklist/
2. Read our blog post about "Submitting Great PRs" https://developers.mattermost.com/blog/2019-01-24-submitting-great-prs
3. Take a look at other repository specific documentation at https://developers.mattermost.com/contribute
-->

#### Summary
As a beginner, it's somtimes becomes difficult to ice break the jargon path of installation and hence contribution. Hence, catering to beginners I've added a deatiled steps to help beginners to install and explore the project, irrespective of OS.

#### Ticket Link
NA

mattermod commented:
Hello @helios1101,

Thanks for your pull request! A Core Committer will review your pull request soon. For code contributions, you can learn more about the review process [here](https://developers.mattermost.com/contribute/getting-started/code-review/).

helios1101 commented:
Hi,
I'm facing below error in build. 
Can anyone guide me how it can be resolved.

```
1324
  remote: Permission to mattermost/docs.git denied to github-actions[bot].
1325
  fatal: unable to access 'https://github.com/mattermost/docs.git/': The requested URL returned error: 403
1326
  Error: Action failed with "The process '/usr/bin/git' failed with exit code 128"
```

justinegeffen commented on README.md:
```diff
@@ -55,13 +57,30 @@ Any merged PRs with an **Editor Review** or **Reviews Complete** label will be p
-2. Install [pipenv](https://docs.pipenv.org/): `pip install pipenv`. 
-3. `cd` into the cloned repository.
-4. Install the required packages: `pipenv install`.
+If you've downloaded the repo and are editing Mattermost documentation on your local machine, you can generate the HTML files from markdown in the `/source` directory. You can review them before you commit changes or create pull requests.
```
This is really helpful, thank you! I really appreciate having some detailed steps around this process. 

However, IIRC, installing pip on macOS requires a different command and/or can be installed using Homebrew which is painless. Is it possible to confirm that these steps have been performed and validated on Linux, macOS, and Windows? If not, one option is to separate the three OSs and have individual processes for them.

helios1101 replied:
Hi @justinegeffen 
This steps are working on Windows and Linux. Yes, for Mac `HomeBrew` comes to rescue.
Nevertheless, rest of the followed commands remains unchanged.

justinegeffen commented on README.md:
````diff
@@ -55,13 +57,30 @@ Any merged PRs with an **Editor Review** or **Reviews Complete** label will be p
+
+1. Download the Mattermost docs repo onto a machine with Python installed or simple cloned the forked copy of this repo by running,
+```sh
+$ git clone https://github.com/Your_Username/docs.git
````
What would be the difference between cloning the docs repo and cloning your fork? The two URLs are different so if both options are provided it might be a consideration to provide the docs repo URL (https://github.com/mattermost/docs.git).

helios1101 replied:
Sure, updating to as recomended.
Thanks

justinegeffen commented:
Hi @helios1101. Thanks for your PR, it's very helpful! I've added some comments and feedback.

With regard to the build error, it's most likely related to our preview server which spins up a preview site for PRs. However, it's not able to do so with a forked repo which results in an error. It doesn't affect your PR though, so no need to worry. :)

helios1101 commented:
Hi @justinegeffen 
Updated the commits according to feedbacks and comments.
Please have a look.

justinegeffen commented on README.md:
```diff
@@ -55,13 +57,33 @@ Any merged PRs with an **Editor Review** or **Reviews Complete** label will be p
-4. Install the required packages: `pipenv install`.
+If you've downloaded the repo and are editing Mattermost documentation on your local machine, you can generate the HTML files from markdown in the `/source` directory. You can review them before you commit changes or create pull requests.
+
+`NOTE: Commands are to be executed on Linux, Mac, and Windows(using Powershell)`
```
```suggestion
**Note:** Commands can be executed on Linux, Mac, and Windows (using Powershell)
```

helios1101 replied:
Updated !

justinegeffen commented on README.md:
```diff
@@ -55,13 +57,33 @@ Any merged PRs with an **Editor Review** or **Reviews Complete** label will be p
+
+`NOTE: Commands are to be executed on Linux, Mac, and Windows(using Powershell)`
+
+1. Download the Mattermost docs repo onto a machine with Python installed or simple cloned the forked copy of this repo by running,
```
```suggestion
1. Clone the forked copy of the docs repo by running:
```

helios1101 replied:
Updated !

justinegeffen commented on README.md:
````diff
@@ -55,13 +57,33 @@ Any merged PRs with an **Editor Review** or **Reviews Complete** label will be p
+```sh
+$ git clone https://github.com/mattermost/docs.git
+```
+2. Install [pipenv](https://docs.pipenv.org/) by following command
````
```suggestion
2. Install [pipenv](https://docs.pipenv.org/) using:
```

justinegeffen replied:
Does this need to be installed within the cloned repo or anywhere on one's computer?

helios1101 replied:
It can be installed anywhere, as what I did.

justinegeffen commented on README.md:
````diff
@@ -55,13 +57,33 @@ Any merged PRs with an **Editor Review** or **Reviews Complete** label will be p
+```
+2. Install [pipenv](https://docs.pipenv.org/) by following command
+```sh
+$ brew install pipenv   # for Mac users, if you have brew installed)
````
```suggestion
$ brew install pipenv (macOS)
```

helios1101 replied:
I think of overriding this suggestion and keeping the change, as this will help people to gain more clarity on what commands.

justinegeffen replied:
Thanks, @helios1101! That's a good point. How about something like: "(if you don't have Homebrew installed, download it here") or similar?

justinegeffen commented on README.md:
````diff
@@ -55,13 +57,33 @@ Any merged PRs with an **Editor Review** or **Reviews Complete** label will be p
+$ brew install pipenv   # for Mac users, if you have brew installed)
+```
+```python
+$ pip install pipenv # for all others
````
```suggestion
$ pip install pipenv (other operating systems)
```

helios1101 replied:
Updated. Thanks :)

justinegeffen commented on README.md:
````diff
@@ -55,13 +57,33 @@ Any merged PRs with an **Editor Review** or **Reviews Complete** label will be p
+```python
+$ pip install pipenv # for all others
+```
+3. Navigate into the cloned repository by following command,
````
```suggestion
3. Navigate into the cloned repository:
```

justinegeffen replied:
Should we also include steps around how to find the built files using Finder/Explorer? Or does this step allow the contributor to select a file and view in a browser?

helios1101 replied:
I think this was enough to navigate, as changing directory by `cd` is what is the correct expectations.

justinegeffen commented:
Thanks, @helios1101. This is a really helpful update. I do have a few questions:

1. Cloning the Mattermost docs repo won't include any changes made by a contributor. It might be an idea to clarify that you need to clone your fork to see your changes.
2. Should we include instructions about how to edit files locally or are these just steps to preview changes from your fork? The reason this information hasn't been included is that it can add a layer of complexity and barrier to entry for new contributors. The easiest way to contribute is to fork the repo > make changes > view in preview > commit > push. 
3. That said, I am very excited to encourage contributors to edit their changes locally and view a preview. If we do that, should we also include links to editors that support ``.rst`` authoring? Or can we assume that the contributor is using one already?

Thanks again for your work and contribution. This is an exciting change. :)

mattermod commented:
This PR has been automatically labelled "stale" because it hasn't had recent activity.
A core team member will check in on the status of the PR to help with questions.
Thank you for your contribution!

/cc @jasonblais @jfrerich @emilyacook

helios1101 commented:
Hi, I'm working on the comments just got engulfed in some personal work.
Will target to complete this asap with comments.

jasonblais commented:
Sounds good, thanks for the update @helios1101!

helios1101 commented:
Due to regional festivals and occasions the work is delayed. 
Nevertheless, this is there on my front mind.

mattermod commented:
This PR has been automatically labelled "stale" because it hasn't had recent activity.
A core team member will check in on the status of the PR to help with questions.
Thank you for your contribution!

/cc @jasonblais @jfrerich @emilyacook

helios1101 commented:
Raised the pull with my understandings, let me know if any changes are required.
Thanks for being helpful throughout ! :)

justinegeffen commented on README.md:
```diff
@@ -27,7 +27,9 @@ If this is your first time contributing to Mattermost, first read the [Mattermos
 
-If this is the first time you're contributing, click **Edit the file in your fork of this project** (pencil icon) on the top right corner to create a fork.
+If this is the first time you're contributing, follow the steps: 
+1. Fork the repo by clicking on <a href="https://github.com/mattermost/docs/fork?fragment=1"><kbd><b>Fork</b></kbd></a> button.
```
```suggestion
1. Select **Fork** in the top-right corner of the GitHub page to fork the repo.
```

justinegeffen commented on README.md:
```diff
@@ -27,7 +27,9 @@ If this is your first time contributing to Mattermost, first read the [Mattermos
-If this is the first time you're contributing, click **Edit the file in your fork of this project** (pencil icon) on the top right corner to create a fork.
+If this is the first time you're contributing, follow the steps: 
+1. Fork the repo by clicking on <a href="https://github.com/mattermost/docs/fork?fragment=1"><kbd><b>Fork</b></kbd></a> button.
+2. Navigate to file you would like to change and Click **Edit the file** (pencil icon) on the top right corner to start editing in Github UI.
```
```suggestion
2. Navigate to file you want to edit and select the pencil icon (**Edit the file**) to open the editing interface.
```

justinegeffen commented on README.md:
````diff
@@ -55,13 +57,33 @@ Any merged PRs with an **Editor Review** or **Reviews Complete** label will be p
+$ brew install pipenv   # for Mac users, if you have brew installed)
+```
+```python
+$ pip install pipenv # for other operating systems)
````
```suggestion
$ pip install pipenv (for other operating systems)
```

justinegeffen commented:
Thanks, @helios1101! This is looking really good! I've added some feedback and I think we're getting close to merging this. I would like to get some feedback from @cwarnermm before that, though, as she's just been through this process and may have some more content to add.

helios1101 commented:
> Thanks, @helios1101! This is looking really good! I've added some feedback and I think we're getting close to merging this. I would like to get some feedback from @cwarnermm before that, though, as she's just been through this process and may have some more content to add.

Sure, I've added your suggestions ! 
Looking forward to merge soon :)

cwarnermm commented on README.md:
````diff
@@ -55,13 +57,33 @@ Any merged PRs with an **Editor Review** or **Reviews Complete** label will be p
+
+1. Open a terminal window and clone the forked copy of the docs repo by running:
+```sh
+$ git clone https://github.com/mattermost/docs.git
````
Including $ characters at the beginning of CLI entries may suggest to the reader that the $ is required (when it's not). I recommend removing these instances for clarity.

helios1101 replied:
I think whever we open a tereminal we have these symbols that could help people understand that we need to run exactly these commands. 
Not sure if we need to remove these.

cwarnermm replied:
@justinegeffen - I'd like your perspective on this.

justinegeffen replied:
We don't specify this in the style guide so I am 0/5 on which we want to go with provided we pick one and then apply it consistently.

cwarnermm replied:
If our goal is to provide command line entries that can be easily copied and pasted directly from docs to a terminal, I recommend that we remove them to avoid cases where the coped commands include the command prompt symbol. 

Further, because the command prompt symbol is often configurable, different users may see different symbols than what is documented. I'm 3/5 for removing the symbols and including this rationale in the Style Guide for omitting them in documented commands.

helios1101 replied:
Hi @cwarnermm and @justinegeffen 
I think we are then moving forward with removing the `$` symbol. Thanks for the feedbacks.

cwarnermm commented on README.md:
````diff
@@ -55,13 +57,33 @@ Any merged PRs with an **Editor Review** or **Reviews Complete** label will be p
+```
+2. Install [pipenv](https://docs.pipenv.org/) by using:
+```sh
+$ brew install pipenv   (for Mac users, if you have brew installed)
````
To ensure clarity, reference "brew" as "Homebrew".

In addition, consider moving the platform/operating system references from the end of the sentence to the beginning to make them more prominent/visible.

helios1101 replied:
Sure, updated.

cwarnermm commented on README.md:
```diff
@@ -27,7 +27,9 @@ If this is your first time contributing to Mattermost, first read the [Mattermos
 The quickest way to begin is editing directly on GitHub on your fork of the Mattermost docs repo. Click the **Edit** icon on the top right corner of the page you want to edit in the Mattermost documentation.
 
-If this is the first time you're contributing, click **Edit the file in your fork of this project** (pencil icon) on the top right corner to create a fork.
+If this is the first time you're contributing, follow the steps: 
```
Consider changing "follow the steps:" to "follow _these_ steps:"

helios1101 replied:
Sure, updated.

cwarnermm reviewed (changes requested):
Excellent clarifications! I've provided additional updates to consider.

helios1101 commented on README.md:
````diff
@@ -55,13 +57,33 @@ Any merged PRs with an **Editor Review** or **Reviews Complete** label will be p
+6. Navigate to the `/build` directory to preview the page/s you've edited, by runnning 
+```sh
+$ cd /build
+```
````
Not sure, but I personally feel navgating in `build` directory should suffice. In-case you insist I can add `run index.html`

cwarnermm replied:
@justinegeffen - I'd love to hear your thoughts on this!

justinegeffen replied:
I've never used that command actually! I use `make html` and then open a file from the docs > build directory to view in a browser. How does `run index.html` work in relation to that?

cwarnermm replied:
Once you've built the documentation set, you review the documentation by running the `index.html` file in a browser to open the full documentation set and review your changes. 

If we agree that our target audiences understand how to run the doc set by running `index.html`, then I'm 3/5 for leaving this level of detail out. However, if including this level of detail helps lower the barrier to entry for more documentation contributors, I'm 4/5 for including it.

helios1101 replied:
Hi @cwarnermm and @justinegeffen 
I personally feel this level of detail will suffice. Hence, overriding this change.

helios1101 commented:
Hi @cwarnermm and @justinegeffen 
I've added my points in the your comments. 
In case you feel the necessity to add in that way, will be happy to do that. 
Looking forward for this change merge :)

cwarnermm reviewed (commented):
Open questions for @justinegeffen inline.

justinegeffen commented:
> Open questions for @justinegeffen inline.

Thanks! I've provided some feedback on the two outstanding comments. :)

cwarnermm commented:
I've provided feedback inline for the two outstanding comments as well :)

cwarnermm reviewed (commented):
Thank you @helios1101 for your patience with this PR.

mattermod commented:
This PR has been automatically labelled "stale" because it hasn't had recent activity.
A core team member will check in on the status of the PR to help with questions.
Thank you for your contribution!

/cc @jasonblais @jfrerich @emilyacook

helios1101 commented:
Hi, 
I'm on vacation leave for a week, post which I'll address the comments.

mattermod commented:
This PR has been automatically labelled "stale" because it hasn't had recent activity.
A core team member will check in on the status of the PR to help with questions.
Thank you for your contribution!

/cc @jasonblais @jfrerich @emilyacook

jfrerich commented:
Hi @helios1101! I'm just checking in to see if you have a timeframe for completing this ticket? It would be great to get this merged!

helios1101 commented:
> Hi @helios1101! I'm just checking in to see if you have a timeframe for completing this ticket? It would be great to get this merged!

Sure, will resume working on this.

helios1101 commented:
Hi @justinegeffen @cwarnermm 
Looking forward to your valuable comments.

cwarnermm commented on README.md:
```diff
@@ -27,7 +27,9 @@ If this is your first time contributing to Mattermost, first read the [Mattermos
 
-If this is the first time you're contributing, click **Edit the file in your fork of this project** (pencil icon) on the top right corner to create a fork.
+If this is the first time you're contributing, follow these steps: 
+1. Select **Fork** in the top-right corner of the GitHub page to fork the repo.
```
```suggestion
1. Select **Fork** in the top-right corner of the GitHub page to fork the repository.
```

cwarnermm commented on README.md:
```diff
@@ -27,7 +27,9 @@ If this is your first time contributing to Mattermost, first read the [Mattermos
-If this is the first time you're contributing, click **Edit the file in your fork of this project** (pencil icon) on the top right corner to create a fork.
+If this is the first time you're contributing, follow these steps: 
+1. Select **Fork** in the top-right corner of the GitHub page to fork the repo.
+2. Navigate to file you want to edit and select the pencil icon (**Edit the file**) to open the editing interface.
```
```suggestion
2. Navigate to file you want to edit, and select the Pencil icon (**Edit the file**) to open the editing interface.
```

cwarnermm commented on README.md:
```diff
@@ -55,13 +57,36 @@ Any merged PRs with an **Editor Review** or **Reviews Complete** label will be p
-2. Install [pipenv](https://docs.pipenv.org/): `pip install pipenv`. 
-3. `cd` into the cloned repository.
-4. Install the required packages: `pipenv install`.
+If you've downloaded the repo and are editing Mattermost documentation on your local machine, you can generate the HTML files from markdown in the `/source` directory. You can review them before you commit changes or create pull requests.
```
```suggestion
If you've downloaded the repository and are editing Mattermost documentation on your local machine, you can generate the HTML files from markdown in the `/source` directory. You can review your changes before you commit them or create pull requests.
```

cwarnermm commented on README.md:
```diff
@@ -55,13 +57,36 @@ Any merged PRs with an **Editor Review** or **Reviews Complete** label will be p
-4. Install the required packages: `pipenv install`.
+If you've downloaded the repo and are editing Mattermost documentation on your local machine, you can generate the HTML files from markdown in the `/source` directory. You can review them before you commit changes or create pull requests.
+
+**Note:** Commands can be executed on Linux, Mac, and Windows (using Powershell)
```
```suggestion
**Note:** Commands can be executed on Linux, Mac, and Windows (using Powershell).
```

cwarnermm commented on README.md:
```diff
@@ -55,13 +57,36 @@ Any merged PRs with an **Editor Review** or **Reviews Complete** label will be p
+
+**Note:** Commands can be executed on Linux, Mac, and Windows (using Powershell)
+
+1. Open a terminal window and clone the forked copy of the docs repo by running:
```
```suggestion
1. Open a terminal window, and clone the forked copy of the documentation repository by running the following command:
```

cwarnermm commented on README.md:
````diff
@@ -55,13 +57,36 @@ Any merged PRs with an **Editor Review** or **Reviews Complete** label will be p
+```python
+pip install pipenv 
+```
+3. Open your terminal and navigate into the cloned repository:
````
```suggestion
3. Open your terminal, and navigate into the cloned repository:
```

cwarnermm commented on README.md:
````diff
@@ -55,13 +57,36 @@ Any merged PRs with an **Editor Review** or **Reviews Complete** label will be p
+```
 5. Build the doc set using `make html`. This generates files in `/build` directory.
-6. Navigate to the `/build` directory to preview the page/s you've edited. 
+6. Navigate to the `/build` directory to preview the page/s you've edited, by runnning 
````
```suggestion
6. Navigate to the `/build` directory to preview your changes by running the following command:
```

cwarnermm commented on README.md:
````diff
@@ -55,13 +57,36 @@ Any merged PRs with an **Editor Review** or **Reviews Complete** label will be p
+```sh
+git clone https://github.com/mattermost/docs.git
+```
+2. Install [pipenv](https://docs.pipenv.org/) by using:     
````
```suggestion
2. Install [pipenv](https://docs.pipenv.org/) by using one of the following commands:     
```

cwarnermm commented on README.md:
````diff
@@ -55,13 +57,36 @@ Any merged PRs with an **Editor Review** or **Reviews Complete** label will be p
+```sh
+cd docs
+```
+4. Install the required packages by running
````
```suggestion
4. Install the required packages by running the following command:
```

cwarnermm reviewed (commented):
Proposals suggested inline.

helios1101 commented:
Hi @cwarnermm 
Done the fix as per changes suggested. 
Looking forward to merge soon.

cwarnermm commented on README.md:
```diff
@@ -55,13 +57,36 @@ Any merged PRs with an **Editor Review** or **Reviews Complete** label will be p
+
+**Note:** Commands can be executed on Linux, Mac, and Windows (using Powershell).
+
+1. Open a terminal window, and clone the forked copy of the documentation repository by running the following command::
```
```suggestion
1. Open a terminal window, then clone a forked copy of the documentation repository by running the following command:
```

cwarnermm commented on README.md:
````diff
@@ -55,13 +57,36 @@ Any merged PRs with an **Editor Review** or **Reviews Complete** label will be p
+```sh
+git clone https://github.com/mattermost/docs.git
+```
+2. Install [pipenv](https://docs.pipenv.org/) by using one of the following commands:
````
```suggestion
2. Install [pipenv](https://docs.pipenv.org/) by using one of the following commands based on your operating system:
```

cwarnermm commented on README.md:
````diff
@@ -55,13 +57,36 @@ Any merged PRs with an **Editor Review** or **Reviews Complete** label will be p
+```
+2. Install [pipenv](https://docs.pipenv.org/) by using one of the following commands:
+
+-> for Mac users, if you have Homebrew installed
````
```suggestion
For Mac users where Homebrew is installed:
```

cwarnermm commented on README.md:
````diff
@@ -55,13 +57,36 @@ Any merged PRs with an **Editor Review** or **Reviews Complete** label will be p
+```sh
+brew install pipenv  
+```
+-> for other operating systems
````
```suggestion
For other operating systems:
```

cwarnermm commented on README.md:
````diff
@@ -55,13 +57,36 @@ Any merged PRs with an **Editor Review** or **Reviews Complete** label will be p
+```python
+pip install pipenv 
+```
+3. Open your terminal, and navigate into the cloned repository:
````
```suggestion
3. In the terminal window, navigate into the cloned repository:
```

cwarnermm commented on README.md:
````diff
@@ -55,13 +57,36 @@ Any merged PRs with an **Editor Review** or **Reviews Complete** label will be p
+```python
+pipenv install
+```
 5. Build the doc set using `make html`. This generates files in `/build` directory.
````
```suggestion
5. Build the documentation set using `make html`. This generates files in `/build` directory.
```

cwarnermm reviewed (commented):
I've included proposed suggestions inline.

helios1101 commented:
Hi @cwarnermm 
Encorporated the suggested changes.
Looking forward to merge.

cwarnermm commented on README.md:
````diff
@@ -55,13 +57,36 @@ Any merged PRs with an **Editor Review** or **Reviews Complete** label will be p
+**Note:** Commands can be executed on Linux, Mac, and Windows (using Powershell).
+
+1. Open a terminal window, then clone a forked copy of the documentation repository by running the following command::
+```sh
````
@justinegeffen - do we need preceding spaces in order for the code-formatted examples to display correctly? If so, will impact the following steps below as well.

justinegeffen replied:
I'll confirm.

justinegeffen replied:
No, it looks like it displays as expected without them.

cwarnermm reviewed (approved):
Formatting question left inline for @justinegeffen

helios1101 commented:
Hi @justinegeffen 
SInce the mattermost site says, first contribution is eligible for some swags was just worried whom to contact (community mail or so) for the same. 
Was too excited for this achievement.

jasonblais commented:
@helios1101 Thanks for your contribution! And indeed, first contributions receive special swag. You can reach us via email at community[at]mattermost.com!

You can expect it to arrive within 2-3 months following the PR merge date (so most likely in April). The shipping time is longer than usual due to the on-going pandemic.

We're also working on a process improvement to notify all first-time contributors about when and how to expect the swag - and also to add it to our handbook. 

Thanks again for your contribution!! :)
